sqldb/migrations.go

```diff
@@ -2,22 +2,111 @@ package sqldb
 
 import (
 	"bytes"
+	"context"
+	"database/sql"
 	"errors"
+	"fmt"
 	"io"
 	"io/fs"
 	"net/http"
+	"sort"
 	"strings"
+	"time"
 
 	"github.com/btcsuite/btclog/v2"
 	"github.com/golang-migrate/migrate/v4"
 	"github.com/golang-migrate/migrate/v4/database"
 	"github.com/golang-migrate/migrate/v4/source/httpfs"
+	"github.com/lightningnetwork/lnd/sqldb/sqlc"
 )
 
+var (
+	// migrationConfig defines a list of migrations to be applied to the
+	// database. Each migration is assigned a version number, determining
+	// its execution order.
+	// The schema version, tracked by golang-migrate, ensures migrations are
+	// applied to the correct schema. For migrations involving only schema
+	// changes, the migration function can be left nil. For custom
+	// migrations an implemented migration function is required.
+	//
+	// NOTE: The migration function may have runtime dependencies, which
+	// must be injected during runtime.
+	migrationConfig = []MigrationConfig{
+		{
+			Name:          "000001_invoices",
+			Version:       1,
+			SchemaVersion: 1,
+		},
+		{
+			Name:          "000002_amp_invoices",
+			Version:       2,
+			SchemaVersion: 2,
+		},
+		{
+			Name:          "000003_invoice_events",
+			Version:       3,
+			SchemaVersion: 3,
+		},
+		{
+			Name:          "000004_invoice_expiry_fix",
+			Version:       4,
+			SchemaVersion: 4,
+		},
+		{
+			Name:          "000005_migration_tracker",
+			Version:       5,
+			SchemaVersion: 5,
+		},
+		{
+			Name:          "000006_invoice_migration",
+			Version:       6,
+			SchemaVersion: 6,
```

Comment:
So the `SchemaVersion` is the version we'll migrate the schema _first_ and any optional in-code migration will be applied on top of that. See this snippet: https://github.com/bhandras/lnd/blob/18091a75d2adc05a0a0ecaf53898d9c79186aaff/sqldb/migrations.go#L367

As a concrete example: 
```
{
			Name:          "000006_invoice_migration",
			Version:       6,
			SchemaVersion: 6,
},
```

In this migration, the schema is first migrated to "schema version" 6 by applying the contents of the file `000006_invoice_migration.up.sql`. After that, if the user has requested runtime migration, the migration function will be attached, and the overall version will be updated to 6. See the relevant code here: https://github.com/bhandras/lnd/blob/18091a75d2adc05a0a0ecaf53898d9c79186aaff/config_builder.go#L1118).

I totally agree that we could split this up to a separate version 7 for better readability.

Reply:
yeah gotcha re how it is working. 
Just personal preference would be 1 logical migration per config since as is, it is possible to have "Version 6" run but then actually only the schema update succeeds and then the code-level mig fails and then when talking about it, we would need to say "the second part of the migration 6 (ie, the code part) failed" but part of the migration succeeded. I think it's nicer to say "mig 6 completed, mig 7 failed"

but yeah - defs a subjective opinion i guess. Happy to let second reviewer weigh in

Reply:
another point for the suggestion is: if we are are reading the code later on and want to see where a migration was introduced and we go back to this commit: I rate we want to see the entire migration in the commit. Currently this commit only tells half the story of mig 6